numpy/ma/core.py

```diff
@@ -1296,14 +1296,23 @@ def _recursive_make_descr(datatype, newtype=bool_):
                 # Prepend the title to the name
                 name = (field[-1], name)
             descr.append((name, _recursive_make_descr(field[0], newtype)))
-        return descr
+        newtype = np.dtype(descr)
+
     # Is this some kind of composite a la (np.float,2)
     elif datatype.subdtype:
         mdescr = list(datatype.subdtype)
         mdescr[0] = _recursive_make_descr(datatype.subdtype[0], newtype)
-        return tuple(mdescr)
+        newtype = np.dtype(tuple(mdescr))
+
+    # promote the type to a dtype if otherwise untouched
     else:
-        return newtype
```

Comment:
I think that would be better, but it is a very mild preference at best, so also fine to leave it. For easy following of the flow of logic, I would still think it is best to not reuse the `newtype` name, though (it really threw me off, to convince me that the recursive calls could not pick that up).